lmcache/storage_backend/local_backend.py

```diff
@@ -126,31 +126,58 @@ def put_nonblocking(self, key, kv_chunk):
         for evict_key in evict_keys:
             self.remove(evict_key)
 
+        # free old block to avoid mem leak
+        if key in self.dict:
+            self.remove(key)
+
+        # Allocate the kv chunk
+        kv_obj = self.mpool.allocate(kv_chunk)
+        self.update_lock.release()
+
+        if kv_obj is None:
+            return
+
+        put_stream = torch.cuda.Stream()
+        if kv_chunk.device != torch.cpu:
+            put_stream.wait_stream(torch.cuda.default_stream(kv_chunk.device))
```

Comment:
Not sure about this: do we need to `wait_stream` here, or do it at the beginning of `batched_put`?

Reply:
Actually, I feel like the local CPU Backend does not need a separate thread. Let's talk about this tomorrow?

Reply:
Updated the comment. The wait is needed as there are operations done on that chunk by the main stream. Will try to remove the extra threads.